VideoWeb/VideoWeb/Helpers/ConsultationInvitationTracker.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using VideoWeb.Common.Caching;
+using VideoWeb.Common.Models;
+
+namespace VideoWeb.Helpers
+{
+    public interface IConsultationInvitationTracker
+    {
+        Task<Guid> StartTrackingInvitation(Conference conference, string roomLabel, Guid requestedParticipantId);
+        Task<ConsultationInvitation> GetInvitation(Guid invitationId);
+        Task UpdateConsultationResponse(Guid invitationId, Guid participantId, ConsultationAnswer answer);
+        Task<bool> HaveAllParticipantsAccepted(Guid invitationId);
+        Task<bool> HaveAllParticipantsResponded(Guid invitationId);
+    }
+
+    public class ConsultationInvitationTracker : IConsultationInvitationTracker
+    {
+        private readonly IConsultationInvitationCache _consultationInvitationCache;
+
+        public ConsultationInvitationTracker(IConsultationInvitationCache consultationInvitationCache)
+        {
+            _consultationInvitationCache = consultationInvitationCache;
+        }
+
+        public async Task<Guid> StartTrackingInvitation(Conference conference, string roomLabel, Guid requestedParticipantId)
+        {
+            var requestedParticipant = conference.Participants.FirstOrDefault(p => p.Id == requestedParticipantId);
+
+            if (requestedParticipant == null) 
+                return Guid.Empty;
+            
+            var consultationInvitation = ConsultationInvitation.Create(requestedParticipantId, roomLabel, requestedParticipant.LinkedParticipants.Select(linkedParticipant => linkedParticipant.LinkedId));
+            await _consultationInvitationCache.Write(consultationInvitation);
+            return consultationInvitation.InvitationId;
+        }
+
+        public async Task<ConsultationInvitation> GetInvitation(Guid invitationId)
+        {
+            if (invitationId == Guid.Empty)
```

Comment:
does this mean all VHO invites are treated as one entity? would this not impact other hearings since the cache is for the platform all VHO invites are managed the same way?

Reply:
No because the cache is based on a unique GUID per invite. The actual invitation doesn't even track who requested it just the state of those that were requested.